src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -171,18 +170,18 @@ public Task ExecuteChildContentAsync()
         /// <returns>A <see cref="Task"/> that on completion returns the rendered child content.</returns>
         /// <remarks>
         /// Child content is only executed once. Successive calls to this method or successive executions of the 
-        /// returned <see cref="Task{string}"/> return a cached result.
+        /// returned <see cref="Task{TagHelperContent}"/> return a cached result.
         /// </remarks>
-        public async Task<string> GetChildContentAsync()
+        public async Task<TagHelperContent> GetChildContentAsync()
         {
             if (_childContent == null)
             {
-                _startWritingScope();
+                _startTagHelperWritingScope();
                 await _executeChildContentAsync();
-                _childContent = _endWritingScope().ToString();
+                _childContent = _endTagHelperWritingScope();
             }
 
-            return _childContent;
+            return new DefaultTagHelperContent().SetContent(_childContent);
```

Comment:
You'll have to add a bit more to make the copying work as expected/performantly. The generated code that calls into this method looks like:

```C#
if (__tagHelperExecutionContext.Output.IsContentModified)
{
	Write(__tagHelperExecutionContext.Output.GenerateContent());
}
else if (__tagHelperExecutionContext.ChildContentRetrieved)
{
	Write(__tagHelperExecutionContext.GetChildContentAsync().Result);
}
else
{
	__tagHelperExecutionContext.ExecuteChildContentAsync().Wait();
}
```

So in the case that the user retrieved the child content: `__tagHelperExecutionContext.ChildContentRetrieved` you will end up copying the result again before writing it (no bueno). 

Now it's also important to re-copy each time the `GetChildContentAsync` method is called so multiple `TagHelper` authors can see the original content (without worrying about someone else modifying it); you have this part covered.

You'll need to add a capability to retrieve the original, uncopied version for the page to use in the `else if (__tagHelperExecutionContext.ChildContentRetrieved)` clause.